lib/handlers/PostHandler.js

```diff
@@ -1,46 +1,73 @@
 'use strict';
 
 const BaseHandler = require('./BaseHandler');
+const File = require('../models/File');
+const Uid = require('../models/Uid');
 const RequestValidator = require('../validators/RequestValidator');
-const ERRORS = require('../constants').ERRORS;
-const EVENT_ENDPOINT_CREATED = require('../constants').EVENT_ENDPOINT_CREATED;
+const { EVENTS, ERRORS } = require('../constants');
 const debug = require('debug');
 const log = debug('tus-node-server:handlers:post');
 class PostHandler extends BaseHandler {
+
+    constructor(store, options) {
+        if (options.namingFunction && typeof options.namingFunction !== 'function') {
+            throw new Error('\'namingFunction\' must be a function');
+        }
+
+        if (!options.namingFunction) {
+            options.namingFunction = Uid.rand;
+        }
+
+        super(store, options);
+    }
+
     /**
      * Create a file in the DataStore.
      *
      * @param  {object} req http.incomingMessage
      * @param  {object} res http.ServerResponse
      * @return {function}
      */
-    send(req, res) {
+    async send(req, res) {
         if ('upload-concat' in req.headers && !this.store.hasExtension('concatentation')) {
-            return Promise.resolve(super.send(res, 501, {}, 'Concatenation extension is not (yet) supported. Disable parallel uploads in the tus client. '));
+            throw ERRORS.UNSUPPORTED_CONCATENATION_EXTENSION;
+        }
+
+        const upload_length = req.headers['upload-length'];
+        const upload_defer_length = req.headers['upload-defer-length'];
+        const upload_metadata = req.headers['upload-metadata'];
+
+        if ((upload_length === undefined) === (upload_defer_length === undefined)) {
```

Comment:
```suggestion
        if (!upload_length && !upload_defer_length) {
```

Reply:
This is not the same. Previous code does XOR and checks that either one is defined but not both.